multinode-demo/start_nodes.sh

```diff
@@ -68,40 +66,60 @@ for ip_addr in "${ip_addr_array[@]}"; do
 
   # Stop current nodes
   ssh "$remote_user@$ip_addr" 'pkill -9 solana-'
+}
 
-  if [[ -n $leader ]]; then
-    echo "Adding known hosts for $ip_addr"
-    ssh -n -f "$remote_user@$ip_addr" "ssh-keygen -R $leader"
-    ssh -n -f "$remote_user@$ip_addr" "ssh-keyscan $leader >> ~/.ssh/known_hosts"
+leader() {
+  common_setup "$1"
 
-    ssh -n -f "$remote_user@$ip_addr" "rsync -vPrz ""$remote_user@$leader"":~/.cargo/bin/solana* ~/.cargo/bin/"
-    ssh -n -f "$remote_user@$ip_addr" "rsync -vPrz ""$remote_user@$leader"":~/solana/multinode-demo ~/solana/"
-    ssh -n -f "$remote_user@$ip_addr" "rsync -vPrz ""$remote_user@$leader"":~/solana/fetch-perf-libs.sh ~/solana/"
-  else
-    # Deploy build and scripts to remote node
-    rsync -vPrz ~/.cargo/bin/solana* "$remote_user@$ip_addr":~/.cargo/bin/
-    rsync -vPrz ./multinode-demo "$remote_user@$ip_addr":~/solana/
-    rsync -vPrz ./fetch-perf-libs.sh "$remote_user@$ip_addr":~/solana/
-  fi
+  rsync -vPrz ~/.cargo/bin/solana* "$remote_user@$ip_addr":~/.cargo/bin/ # Deploy build and scripts to remote node
+  rsync -vPrz ./multinode-demo "$remote_user@$ip_addr":~/solana/
+  rsync -vPrz ./fetch-perf-libs.sh "$remote_user@$ip_addr":~/solana/
 
   # Run setup
   ssh "$remote_user@$ip_addr" "$ssh_command_prefix"' ./multinode-demo/setup.sh -p "$ip_addr"'
 
+  echo "Starting leader node $ip_addr"
+  ssh -n -f "$remote_user@$ip_addr" 'cd solana; ./fetch-perf-libs.sh'
+  ssh -n -f "$remote_user@$ip_addr" "$ssh_command_prefix"' SOLANA_CUDA=1 ./multinode-demo/leader.sh > leader.log 2>&1'
+  ssh -n -f "$remote_user@$ip_addr" "$ssh_command_prefix"' ./multinode-demo/drone.sh > drone.log 2>&1'
+  leader_ip=${ip_addr_array[0]}
+}
+
+validator() {
+  common_setup "$1"
+
+  echo "Adding known hosts for $ip_addr"
```

Comment:
Would be nice to merge all this sshing into a single script, like what refresh-testnet.sh does

Reply:
Agree. I can do that in the next iteration.